lgc/tool/lgc/CMakeLists.txt

```diff
@@ -45,7 +45,7 @@ add_llvm_tool(lgc
 
 # lgc is linked in separately to account for both static and dynamic library
 # builds.
-llvm_map_components_to_libnames(extra_llvm_libs lgc)
+llvm_map_components_to_libnames(extra_llvm_libs lgc LgcCps)
```

Comment:
The dependency here is not needed at all is it? You might need to modify main() in lgc.cpp to only set the target and data layout if they are not already set; a test for LgcCps stuff can be derived from -emit-lgc and so would already have the right data layout.

Reply:
Without this, the compiler would complain:
/home/ruiling/projects/vkdriver/drivers/llpc/lgc/tool/lgc/lgc.cpp:32:10: fatal error: lgccps/LgcCpsDialect.h: No such file or directory
We have both tests generated from -emit-lgc and manually written tests. Not assuming the data layout was there in the test make hand writing tests easy.

Reply:
Hmm, maybe what we need to do is to add an LGC API to get the data layout string it wants (which is now different to the data layout string associated with the target). That would then be used by front-ends including the lgc tool, and those front-ends would not need to depend on LgcCps.